ui/src/data/source.tsx

```diff
@@ -479,20 +479,40 @@ export class BackendUnderlaySource implements UnderlaySource {
       entity.idAttribute
     );
 
-    const res = await parseAPIError(
-      this.underlaysApi.listInstances({
-        entityName: entity.name,
-        underlayName: this.underlay.name,
-        query: this.makeQuery(ra, entityId, cohort, conceptSet, limit),
-      })
-    );
+    let pageMarker: string | undefined;
+    let sql: string | undefined;
+    const data: DataEntry[] = [];
+    while (true) {
+      const res = await parseAPIError(
+        this.underlaysApi.listInstances({
```

Comment:
Does this fetch all the rows, all pages? If so, do you think we should use a limit here instead, so the page doesn't time out if there are e.g. lots of notes? I think you could implement that with a very large page size and a single API call.

Reply:
Yes. Doesn't fetching in pages make it less likely to timeout, since each individual fetch is smaller? Is there no backend limit on the page size?

Reply:
Does the page display while fetching or wait for all fetches to finish first? I think you're right about fetching in pages being less likely for the API calls to timeout. I was thinking that the display would time out for fetching all the rows, regardless of whether you spread it out across multiple API calls or not.

RE backend limit on page size: There's nothing in Tanagra, and I think the BQ limit is in total size of the data (20MB across multiple rows, or 100MB in a single row), not number of rows per se.

Reply:
The timeout is only an issue on individual network requests. The spinner can be displayed for as long as necessary.

I was assuming there would be a page size limit but if not I could do one large fetch instead. It's a little questionable though since I'm surprised 10000 wasn't enough already if there isn't a lower limit we're hitting somewhere.

Reply:
Sorry, the approval updated automatically but not your new comment. It will wait until all of the calls return before rendering but it won't fail because the timeout is for each request individually. I should page anyways since the number is arbitrary but we can tweak the page size as needed.